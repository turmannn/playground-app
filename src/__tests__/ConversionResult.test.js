import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import ConversionResult from '../components/conversionResult'

configure({adapter: new Adapter()});
const each  = require("jest-each").default;

describe('conversionResult', () => {
  each([
    ['100', null],
    [null, '$'],
    [null, null]
  ]).it('renders no result', (amount, targetCur) => {
    // when
    const component = shallow(
      <ConversionResult result={amount} targetCurrency={targetCur}/>
    )

    //then
    expect(component.getElements()).toMatchSnapshot();
  })

  it('renders conversion result', () => {
    // when
    const component = shallow(
      <ConversionResult result = '100' targetCurrency='$' />
    )

    //then
    expect(component.getElements()).toMatchSnapshot();
  })
})
