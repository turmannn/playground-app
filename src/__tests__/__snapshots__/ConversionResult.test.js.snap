// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversionResult renders conversion result 1`] = `
Array [
  <h3>
    conversion result: 
    100 $
  </h3>,
]
`;

exports[`conversionResult renders no result 1`] = `
Array [
  <h3>
    conversion result: 
    
  </h3>,
]
`;

exports[`conversionResult renders no result 2`] = `
Array [
  <h3>
    conversion result: 
    
  </h3>,
]
`;

exports[`conversionResult renders no result 3`] = `
Array [
  <h3>
    conversion result: 
    
  </h3>,
]
`;
